scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/synchronization/AOSynchronization.java

```diff
@@ -0,0 +1,824 @@
+                                                 StackTraceUtil.getStackTrace(e));
+        }
+
+        return answer;
+    }
+
+    @Override
+    public <T extends Serializable> T merge(String originator, TaskId taskid, String channel, String key, T value,
+            String remappingFunction) throws InvalidChannelException, CompilationException, IOException {
+        T answer;
+        try {
+            answer = ((Channel<T>) getChannel(channel)).merge(key,
+                                                              value,
+                                                              (BiFunction<? super T, ? super T, ? extends T>) evaluateClosure(remappingFunction,
+                                                                                                                              BiFunction.class));
+            commitIfNeeded(channel);
+            logWithContext(taskid,
+                           channel,
+                           "Run merge on key '" + key + "', with value " + value + " result is: " + answer,
+                           Level.DEBUG);
+        } catch (InvalidChannelException | CompilationException | IOException e) {
+            throw e;
+        } catch (Throwable e) {
+            throw new ClosureEvaluationException("Exception when evaluating closure : " +
+                                                 StackTraceUtil.getStackTrace(e));
+        }
+        return answer;
+    }
+
+    @Override
+    public <T extends Serializable> PredicateActionResult<T> conditionalCompute(String originator, TaskId taskid,
+            String channel, String key, String predicate, String thenRemappingFunction)
+            throws InvalidChannelException, CompilationException, IOException {
+        PredicateActionResult<T> answer;
+        try {
+            Channel<T> chosenChannel = getChannel(channel);
+            T value = chosenChannel.get(key);
+
+            if (evaluateClosure(predicate, BiPredicate.class).test(key, value)) {
+                answer = new PredicateActionResult<>(true,
+                                                     chosenChannel.compute(key,
+                                                                           (BiFunction<? super String, ? super T, ? extends T>) evaluateClosure(thenRemappingFunction,
+                                                                                                                                                BiFunction.class)));
+                commitIfNeeded(channel);
+            } else {
+                answer = new PredicateActionResult<>(false, value);
+            }
+            logWithContext(taskid,
+                           channel,
+                           "Run conditionalCompute on key '" + key + "' result is: " + answer,
+                           Level.DEBUG);
+
+        } catch (InvalidChannelException | CompilationException | IOException e) {
+            throw e;
+        } catch (Throwable e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Catch Exception instead of Throwable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1181)